tests/python_package_test/test_engine.py

```diff
@@ -901,6 +901,25 @@ def test_max_bin_by_feature(self):
         est = lgb.train(params, lgb_data, num_boost_round=1)
         self.assertEqual(len(np.unique(est.predict(X))), 3)
 
+    def test_small_max_bin(self):
+        np.random.seed(0)
```

Comment:
Please reset the seed before exit from this test.
https://github.com/microsoft/LightGBM/blob/04a56010ec94124da9e8c37006278a5a4b34b842/tests/python_package_test/test_engine.py#L652